src/components/site-form.tsx

```diff
@@ -35,6 +43,41 @@ interface SiteFormErrorProps {
 	className?: string;
 }
 
+interface SiteFormBaseProps {
+	className?: string;
+	children?: React.ReactNode;
+	siteName: string;
+	setSiteName: ( name: string ) => void;
+	sitePath?: string;
+	onSelectPath?: () => void;
+	error: string;
+	doesPathContainWordPress?: boolean;
+	isPathInputDisabled?: boolean;
+	onSubmit: ( event: FormEvent ) => void;
+	fileForImport?: File | null;
+	setFileForImport?: ( file: File | null ) => void;
+	onFileSelected?: ( file: File ) => void;
+	fileError?: string;
+}
+
+interface SiteFormWithVersionsProps extends SiteFormBaseProps {
+	allowVersionsChange: true;
+	phpVersion: SupportedPHPVersion;
+	setPhpVersion: ( version: SupportedPHPVersion ) => void;
+	wpVersion: string;
+	setWpVersion: ( version: string ) => void;
+}
+
+interface SiteFormWithoutVersionsProps extends SiteFormBaseProps {
+	allowVersionsChange?: false;
+	phpVersion?: never;
+	setPhpVersion?: never;
+	wpVersion?: never;
+	setWpVersion?: never;
+}
```

Comment:
We might not need this complexity once we [remove the feature flag](https://github.com/Automattic/dotcom-forge/issues/10557). I see we use `SiteForm` in `EditSite`, but that component will be removed in favor of [`EditSiteDetails`](https://github.com/Automattic/studio/blob/bd081d77cafc0db645299bc1223e4e5827c492c3/src/modules/site-settings/edit-site-details.tsx#L13).

So we could make all those props required instead of optionals. We can do it as a separate PR once we've removed the feature flag.

https://github.com/Automattic/studio/blob/bd081d77cafc0db645299bc1223e4e5827c492c3/src/components/content-tab-settings.tsx#L50

Reply:
Sounds good!
I added a TODO comment to remove this once the edit feature is merged.